SinuS/SinuS/SinuSApp.swift

```diff
@@ -6,9 +6,100 @@
 //
 
 import SwiftUI
+import Firebase
+import UserNotifications
+
+class AppDelegate: NSObject, UIApplicationDelegate {
+    let gcmMessageIDKey = "gcm.message_id"
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]? = nil) -> Bool {
+           FirebaseApp.configure()
+
+           Messaging.messaging().delegate = self
+
+           if #available(iOS 10.0, *) {
+             // For iOS 10 display notification (sent via APNS)
+             UNUserNotificationCenter.current().delegate = self
+
+             let authOptions: UNAuthorizationOptions = [.alert, .badge, .sound]
+             UNUserNotificationCenter.current().requestAuthorization(
+               options: authOptions,
+               completionHandler: {_, _ in })
+           } else {
+             let settings: UIUserNotificationSettings =
+             UIUserNotificationSettings(types: [.alert, .badge, .sound], categories: nil)
+             application.registerUserNotificationSettings(settings)
+           }
+
+           application.registerForRemoteNotifications()
+           return true
+       }
+
+    func application(_ application: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable: Any],
+                         fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+          if let messageID = userInfo[gcmMessageIDKey] {
+            print("Message ID: \(messageID)")
+          }
+
+          print(userInfo)
+
+        completionHandler(UIBackgroundFetchResult.newData)
+    }
+}
+
+extension AppDelegate: MessagingDelegate {
+    func messaging(_ messaging: Messaging, didReceiveRegistrationToken fcmToken: String?) {
+
+      let deviceToken: [String: String] = ["token": fcmToken ?? ""]
+        print("Device token: ", deviceToken) // This token can be used for testing notifications on FCM
+    }
+}
+
+@available(iOS 10, *)
+extension AppDelegate: UNUserNotificationCenterDelegate {
+
+  // Receive displayed notifications for iOS 10 devices.
+  func userNotificationCenter(_ center: UNUserNotificationCenter,
+                              willPresent notification: UNNotification,
+    withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+    let userInfo = notification.request.content.userInfo
+
+    if let messageID = userInfo[gcmMessageIDKey] {
+        print("Message ID: \(messageID)")
+    }
+
+    print(userInfo)
```

Comment:
Still necessary for debugging or not anymore? Otherwise add `TODO(LOEHEN)`

Reply:
they are needed